### ISSUE TITLE
CSpell: move the backdrop.dic file to a dedicated repo under https://github.com/backdrop-ops

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6280]
This is a follow-up to recent issues relevant to cspell; mostly related to (or triggered by) #6255.

Instead of adding words to a PR against core, and then having to wait for one of our core committers to become available, we should follow the same practice as with https://github.com/backdrop-ops/phpcs and move the dictionary to a dedicated https://github.com/backdrop-ops/cspell repo. That way, we can establish a team of maintainers for that repo (I offer myself), and things can move independently of core (and faster too I suspect).

We can create a new GitHub group for the maintainers of that repo (I propose to call it `spelling-bees` 😅 ), and when a PR trips on one or more unknown words, people can either ping us in Zulip or in GitHub or file an issue in the queue of that repo, and we can add the words. No need for core commits, nor for the PR to be rebased - just close it and re-open after the word(s) have been added to the remote/independent repo, and the test will automatically pick the new additions. Profit! 😉 

This issue here is because we'll need a PR against core, to update the https://github.com/backdrop/backdrop/blob/1.x/.cspell/cspell.json file and point to the repo instead of a "local" `backdrop.dic` file (same as we are already doing for the drupal dictionary).

So basically change this:
```json
  "dictionaryDefinitions": [
    {
      "name": "backdrop",
      "path": "backdrop.dic"
    },
    {
      "name": "drupal",
      "path": "https://git.drupalcode.org/project/drupal/-/raw/HEAD/core/misc/cspell/dictionary.txt"
    }
  ],
```

...to this instead:
```json
  "dictionaryDefinitions": [
    {
      "name": "backdrop",
      "path": "https://github.com/backdrop-ops/cspell/blob/main/backdrop.dic"
    },
    {
      "name": "drupal",
      "path": "https://git.drupalcode.org/project/drupal/-/raw/HEAD/core/misc/cspell/dictionary.txt"
    }
  ],
```

**Comment by klonos** [comment]
PS: once https://github.com/streetsidesoftware/cspell-action/issues/869 is in for the cspell GitHub Action, we can consider also moving the `cspell.json` file itself to the new repo.

**Comment by jenlampton** [comment]
An update on this: @quicksketch agreed to this approach in today's dev meeting.

>  new GitHub group for the maintainers of that repo (I propose to call it spelling-bees 😅 )

I have crated this group and invited Greg, Tim, and Nate, as well as myself. Anyone else who wants to help maintain this please ping me on Zulip. 

> move the dictionary to a dedicated [backdrop-ops/cspell](https://github.com/backdrop-ops/cspell) repo

Created, and spelling bees now have access.

**Comment by klonos** [comment]
I think that with the various PRs merged in #6302 over the past few months we have reached a good shape now (not that many commonly-encountered CSpell nags in recent PRs). We should start working on this.

**Comment by quicksketch** [comment]
I consolidated the `backdrop.dic` and the remote `drupal.dic` into a single dictionary and remove the unused words in the https://github.com/backdrop-ops/cspell repository.

I'm going to see if I can make a PR to switch over to that repository and still have tests passing.

**Comment by quicksketch** [comment]
PR filed at https://github.com/backdrop/backdrop/pull/5243 that switches over to the new repository.

It seems to work despite no longer avoiding scanning files like jQuery UI and CKEditor 5. That may be because I consolidated with the Drupal dictionary which had a lot more dictionary words added.

This could be a useful improvement for contrib modules or backdropcms.org projects that want to share the same dictionary in their own GitHub action workflows.

**Comment by quicksketch** [comment]
@indigoxela Would you be able to review this one? Having done a lot of work on improving our workflows and PHPCS, I would appreciate your feedback.

**Comment by quicksketch** [comment]
One last potential change. Maybe we should change https://github.com/backdrop-ops/cspell/blob/main/backdrop.dic to `dictionary.txt`? Seems less likely to cause extension format issues with `.txt` and it's clear what it is. And that would match what Drupal did, for what that's worth.

**Comment by indigoxela** [comment]
> Seems less likely to cause extension format issues with .txt and it's clear what it is.

Makes sense. The ".dic" extension seems related to "text/x-c".

**Comment by quicksketch** [comment]
I made that change over in https://github.com/backdrop-ops/cspell/ and re-ran the action against the new `dictionary.txt` file. Seems to pass just fine without any changes to this PR.

**Comment by indigoxela** [comment]
> Would you be able to review this one?

Reviewed and it looks good, RTBC. 👍

**Comment by quicksketch** [comment]
Super. I'll let it sit for a day or two. Thank you @indigoxela!

**Comment by quicksketch** [comment]
I was feeling hasty this morning and went ahead and merged this. I'm trying to get PRs for 1.34.0 completely clean. Thank you @indigoxela for reviewing! Thank you @klonos for the suggestion and @jenlampton for doing the infrastructure work to get this ready for the switch.

**Comment by quicksketch** [comment]
It took a few more commits to get everything clean again. I merged the following additional PRs:

- https://github.com/backdrop/backdrop/pull/5244
- https://github.com/backdrop/backdrop/pull/5245
